mirrord/layer/src/load.rs

```diff
@@ -38,6 +39,17 @@ static BUILD_TOOL_PROCESSES: LazyLock<HashSet<&str>> = LazyLock::new(|| {
     ])
 });
 
+static COMBINED_BUILD_TOOLS: LazyLock<HashSet<String>> = LazyLock::new(|| {
```

Comment:
I think we might not need this to be a global.

[`should_load`](https://github.com/metalbear-co/mirrord/blob/main/mirrord/layer/src/load.rs#L97) could take the `layer_config.skip_extra_build_tools` as an argument, since we have the `LayerConfig` in [`load_type`](https://github.com/metalbear-co/mirrord/blob/8b1d9e1eeb2cb37acaadfcc563f558466d738dbc/mirrord/layer/src/load.rs#L123C9-L123C19). And pass it down to [`is_build_tool`](https://github.com/metalbear-co/mirrord/blob/8b1d9e1eeb2cb37acaadfcc563f558466d738dbc/mirrord/layer/src/load.rs#L86).

You'll also need to modify [`is_sip_only`](https://github.com/metalbear-co/mirrord/blob/8b1d9e1eeb2cb37acaadfcc563f558466d738dbc/mirrord/layer/src/load.rs#L171) to take this config as a function argument, since it calls `is_build_tool`.